nav2_localization/src/plugins/sample_motion_model.cpp

```diff
@@ -19,27 +20,32 @@ bool DiffDriveOdomMotionModel::SampleFrom(
 
     double x_bar_prime = action.curr_odom.transform.translation.x;
     double y_bar_prime = action.curr_odom.transform.translation.y;
-    double theta_bar_prime = tf2::Quaternion(action.curr_odom.transform.rotation).getAngle();
-
+    double theta_bar_prime = tf2::getYaw(action.curr_odom.transform.rotation);
+    
     double x_bar = action.prev_odom.transform.x;
     double y_bar = action.prev_odom.transform.y;
-    double theta_bar = tf2::Quaternion(action.prev_odom.transform.rotation).getAngle();
+    double theta_bar = tf2::getYaw(action.prev_odom.transform.rotation);
 
     double x = prev_pose.transform.translation.x;
     double y = prev_pose.transform.translation.y;
-    double theta = tf2::Quaternion(prev_pose.transform.rotation).getAngle();
+    double theta = tf2::getYaw(prev_pose.transform.rotation);
 
     double delta_rot_1 = atan2(y_bar_prime-y_bar, x_bar_prime-x_bar) - theta_bar;
+    if(isnan(delta_rot_1) || isinf(delta_rot_1))
+    {
+        RCLCPP_ERROR(node_->get_logger(), "deltta_rot_1 is NAN or INF");
+        delta_rot_1 = 0.0; // TODO: consider a different value
+    }
     double delta_trans = sqrt((x_bar_prime-x_bar)*(x_bar_prime-x_bar) + (y_bar_prime-y_bar)*(y_bar_prime-y_bar));
     double delta_rot_2 = theta_bar_prime - theta_bar - delta_rot_1;
 
-    std::normal_distribution<double> delta_rot_1_noise_dist(0.0, sqrt(alpha1_*delta_rot_1 + alpha2_*delta_trans));
+    std::normal_distribution<double> delta_rot_1_noise_dist(0.0, hypot(alpha1_*delta_rot_1 + alpha2_*delta_trans));
     double delta_rot_1_hat = delta_rot_1 - delta_rot_1_noise_dist(generator_);
 
-    std::normal_distribution<double> delta_trans_noise_dist(0.0, sqrt(alpha3_*delta_trans + alpha4_*(delta_rot_1+delta_rot_2)));
+    std::normal_distribution<double> delta_trans_noise_dist(0.0, hypot(alpha3_*delta_trans + alpha4_*(delta_rot_1+delta_rot_2)));
     double delta_trans_hat = delta_trans - delta_trans_noise_dist(generator_);
 
-    std::normal_distribution<double> delta_rot_2_noise_dist(0.0, sqrt(alpha1_*delta_rot_2 + alpha2_*delta_trans));
+    std::normal_distribution<double> delta_rot_2_noise_dist(0.0, hypot(alpha1_*delta_rot_2 + alpha2_*delta_trans));
```

Comment:
This isn't an appropriate use of hypot. It should only be used for sums of squares.

Reply:
Sorry, that was a mistake that I forgot to undo!